### ISSUE TITLE
Final project codes

### ISSUE THREAD

**Issue by huseingharibshah** [huseingharibshah/Final-project#1]
pip install streamlit pandas numpy scikit-learn mediapipe opencv-python matplotlib

tennis_talent_ai/
│
├── app.py                 # اپلیکیشن اصلی
├── model_text.py          # مدل متنی
├── model_match.py         # پیش‌بینی مسابقه
├── utils_pose.py          # پردازش Pose
└── data/
    └── tennis_data.csv

import pandas as pd
from sklearn.ensemble import RandomForestClassifier

def train_text_model():
    data = {
        "age": [15,18,20,25,30,16,22,28],
        "height": [165,175,180,178,170,168,182,176],
        "weight": [55,68,75,72,70,60,78,73],
        "training_days": [2,4,5,6,3,2,5,4],
        "interest": [5,4,5,3,2,4,5,3],
        "talent": [1,1,1,1,0,0,1,0]
    }

    df = pd.DataFrame(data)
    X = df.drop("talent", axis=1)
    y = df["talent"]

    model = RandomForestClassifier()
    model.fit(X, y)
    return model

import cv2
import mediapipe as mp
import numpy as np

mp_pose = mp.solutions.pose

def analyze_pose(video_path):
    cap = cv2.VideoCapture(video_path)
    scores = []

    with mp_pose.Pose() as pose:
        while cap.isOpened():
            ret, frame = cap.read()
            if not ret:
                break

            image = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
            result = pose.process(image)

            if result.pose_landmarks:
                scores.append(1)

    cap.release()
    return min(100, len(scores) * 5)

from sklearn.ensemble import RandomForestClassifier
import pandas as pd

def train_match_model():
    data = {
        "experience": [1,3,5,7,2,4,6],
        "pose_score": [40,60,80,85,50,70,90],
        "winner": [0,0,1,1,0,1,1]
    }

    df = pd.DataFrame(data)
    X = df.drop("winner", axis=1)
    y = df["winner"]

    model = RandomForestClassifier()
    model.fit(X, y)
    return model

import streamlit as st
import pandas as pd
from model_text import train_text_model
from model_match import train_match_model
from utils_pose import analyze_pose
import tempfile

st.set_page_config(page_title="AI Tennis Talent", layout="centered")
st.title("🎾 سیستم استعداد‌یابی تنیس با هوش مصنوعی")

# -------- Stage 1 --------
st.header("مرحله ۱: تحلیل اطلاعات فردی")

age = st.number_input("سن", 10, 40)
height = st.number_input("قد (cm)", 140, 200)
weight = st.number_input("وزن (kg)", 40, 120)
training = st.slider("روزهای تمرین در هفته", 0, 7)
interest = st.slider("علاقه به ورزش‌های راکتی", 1, 5)

if st.button("تحلیل استعداد"):
    model = train_text_model()
    X = [[age, height, weight, training, interest]]
    result = model.predict(X)[0]

    if result == 1:
        st.success("✅ شما استعداد مناسبی برای تنیس دارید")
    else:
        st.warning("❌ پیشنهاد می‌شود ورزش دیگری را بررسی کنید")

# -------- Stage 2 --------
st.header("مرحله ۲: تحلیل حرکت (Pose Detection)")
video = st.file_uploader("آپلود ویدئو حرکت تنیس", type=["mp4"])

if video:
    tfile = tempfile.NamedTemporaryFile(delete=False)
    tfile.write(video.read())
    score = analyze_pose(tfile.name)
    st.info(f"🎯 امتیاز فرم حرکتی: {score}/100")

# -------- Stage 3 --------
st.header("مرحله ۳: پیش‌بینی نتیجه مسابقه")

experience = st.slider("سال‌های تجربه", 0, 10)
pose_score = st.slider("امتیاز Pose", 0, 100)

if st.button("پیش‌بینی مسابقه"):
    model = train_match_model()
    prediction = model.predict([[experience, pose_score]])[0]

    if prediction == 1:
        st.success("🏆 احتمال برد شما بالاست")
    else:
        st.error("📉 احتمال باخت بیشتر است")
